### PR TITLE
[MM-43443] Don't log error when SendEmailNotifications is set to false

### PR DESCRIPTION
**Pull request by lkyuchukov**
#### Summary
Do not log an error when testing the SMTP connection when SendEmailNotifications is false.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/20001

#### Release Note
```
NONE
```

### CODE REVIEW

**Comment by lkyuchukov** [comment]
/release-note-none
--

**Comment by lkyuchukov** [comment]
@agnivade could you review this PR when you have the time?

File: shared/mail/mail.go [hunk 1]
```diff
@@ -208,10 +208,6 @@ func NewSMTPClient(ctx context.Context, conn net.Conn, config *SMTPConfig) (*smt
 }
 
 func TestConnection(config *SMTPConfig) error {
```
Comment (isacikgoz) [inline shared/mail/mail.go:210]: I think this function should not be called if `SendEmailNotifications` is not true at all. You can check that setting on the call site here:

https://github.com/mattermost/mattermost-server/blob/6ab7cd0f1a7fd8b4540830d5fdef6c69e0d14510/app/server.go#L1199

**Review by agnivade — APPROVED** [review]
Thanks !

**Comment by agnivade** [comment]
Hey @lkyuchukov - there's no need to force-push on the PR :) We already squash all the commits while merging.

**Comment by lkyuchukov** [comment]
Oh ok, I didn't know that. Still, no harm I guess. Anyway, that should be the last commit for this PR.

**Pull request merged**